mindmeld/components/entity_recognizer.py

```diff
@@ -134,16 +135,14 @@ def fit(self,
         self.dirty = True
         return True
 
-    def _data_dump_payload(self):
-        return {
+    def _create_and_dump_payload(self, path):
```

Comment:
I can see the abstract method being overwritten here, but instead of this, consider dumping everything by passing it to the super.dump() directly? That way we'll have a single place for dumping all model specific data instead of 3 separate functions.

Reply:
I think if we do that, the _load_ process might bit become convoluted. Meaning, while loading, it would be a good practice if the base class only loads the dumps the path. But once loaded, base class has no idea what to do with the loaded contents. So it has to pass back the loaded contents to the sub-classes. And doing so feels bit unfaithful while using the name `load()`. What do you think?  @ritvikshrivastava